### DISCUSSION TITLE
problem with 'Use fDOG with multiple seed sequences'

### DISCUSSION THREAD

**Discussion by huangziyan11111** [BIONF/fDOG#21] [Q&A]
Hi, I run the step sbout "Use fDOG with multiple seed sequences" with this following code:
fdogs.run --input seed_folder --jobName combined --outpath AB_dir.combined --refspec TURAR@4572@1
, while it came out without results.

The log information is :
PID 3094180
Starting compiling core orthologs...
==> Core compiling finished in 0.971 sec
Searching orthologs for...
AB.one.2.pep
AB.one.pep
==> Ortholog search finished in 1.155 sec
Joining single outputs...
Final fasta file combined.extended.fa not exists or empty!
HERER
{'mainInput': 'combined.phyloprofile', 'fastaInput': 'combined.extended.fa', 'clusterProfile': 'TRUE'}
combined.config.yml
==> fdogs.run finished in 2.160s

In dir 'seed_folder', there are two file (protein sequence), which I can success to run it respectively.
Hope someone could help me, thanks.

**Comment by trvinh** [comment]
Hi @huangziyan11111,
I see no error from the log, except the message `Final fasta file combined.extended.fa not exists or empty!` (sorry for the next 3 lines after that message, I need to remove them from the log output).
Could you please run that command again with these options: `--cpu 1 --silentOff --keep`
```
fdogs.run --input seed_folder --jobName combined --outpath AB_dir.combined --refspec TURAR@4572@1 --cpu 1 --silentOff --keep --force
```
Then, please check if you have a file `combined_singleOutput.tar.gz` in your `AB_dir.combined` output folder?
Best regards,
Vinh

**Reply by trvinh** [reply]
could you please also show me your `fdog.run` command for the single seed file?

**Reply by huangziyan11111** [reply]
Hi, @trvinh, I rerun the command you said (fdogs.run --input seed_folder --jobName combined --outpath AB_dir.combined --refspec TURAR@4572@1 --cpu 1 --silentOff --keep --force), and this is the log information, I guess maybe the fault is related to hmm. Besides, I have this file combined_singleOutput.tar.gz, while it's empty.
PID 419807
Starting compiling core orthologs...
Validity checking....
Check finished in 0.63 sec!
Please wait while the taxonomy database is indexing...
Indexing done in 0.36 sec!
AB_dir.combined/core_orthologs//AB/hmm_dir/AB.hmm does not exist
Validity checking....
Check finished in 0.18 sec!
Please wait while the taxonomy database is indexing...
Indexing done in 0.01 sec!
AB_dir.combined/core_orthologs//AB/hmm_dir/AB.hmm does not exist
==> Core compiling finished in 0.800 sec
Searching orthologs for...

##### AB.one.2.pep

##### AB.one.pep
==> Ortholog search finished in 1.000 sec
Joining single outputs...
AB_dir.combined/AB/AB.extended.fa
AB_dir.combined/AB/AB.extended.fa
Compressing single outputs...
AB_dir.combined/combined.extended.fa not exists or empty!
HERER
AB_dir.combined/combined.extended.fa', 'clusterProfile': 'TRUE'}
AB_dir.combined/combined.config.yml
==> fdogs.run finished in 1.915s

Maybe your could see my dir_structure (most files are empty):
AB_dir.combined/:
![image](https://user-images.githubusercontent.com/35444490/158089203-aa778a5d-e934-4c65-b960-daed342c7d16.png)
![image](https://user-images.githubusercontent.com/35444490/158089230-8fc731c4-f54f-4e22-a60a-d0e3ce3a758d.png)
![image](https://user-images.githubusercontent.com/35444490/158089282-e804c356-4a0c-4f19-80cb-dbf55332b014.png)



And my log infomation of the single seed sequence is this file:
[run.single.log.txt](https://github.com/BIONF/fDOG/files/8241071/run.single.log.txt)

Hope you can figure out the problem, thank you.

**Reply by trvinh** [reply]
Hi @huangziyan11111,
can you replace the dots in your input filenames to "_" and try again? Also, `AB_one.pep` and `AB_one_2.pep` instead of `AB.one.pep` and `AB.one.2.pep`
I think those dots in the filenames caused the problem, that the _correct_ folders in the `core_orthologs` could not be created. The subfolders in the `core_orthologs` folder need to be named exactly the same as the input files (excluded the file extension).
Best,
Vinh

**Reply by trvinh** [reply]
Here is what I've got after renaming the input files. I have used the same command as yours
```
(bionf) vinh@leonard:/share/project/vinh/test/fdog/fdogs/huangziyan$ ls
AB_dir.combined  AB_one_2.pep  AB_one.pep  fdogs_test.log  seed_folder
(bionf) vinh@leonard:/share/project/vinh/test/fdog/fdogs/huangziyan$ ls AB_dir.combined/
combined.config.yml   combined_forward.domains  combined.phyloprofile     combined_singleOutput.tar.gz  runtime_core.txt
combined.extended.fa  combined_log.txt          combined_reverse.domains  core_orthologs                runtime_ortho.txt
(bionf) vinh@leonard:/share/project/vinh/test/fdog/fdogs/huangziyan$ head AB_dir.combined/combined.phyloprofile
geneID	ncbiID	orthoID	FAS_F	FAS_B
AB_one_2	ncbi9606	AB_one_2|HUMAN@9606@1|A0A024R161|1	1.0	1.0
AB_one	ncbi9606	AB_one|HUMAN@9606@1|A0A024R1R8|1	1.0	1.0
AB_one_2	ncbi7719	AB_one_2|CIOIN@7719@1|F7AEF1|1	0.9298299999999999	0.53095
AB_one	ncbi7719	AB_one|CIOIN@7719@1|H2Y2N4|1	0.9941199999999999	0.99442
AB_one_2	ncbi243274	AB_one_2|THEMA@243274@1|Q9WZV3|1	0.64783	0.4382
AB_one_2	ncbi367110	AB_one_2|NEUCR@367110@1|V5IP41|1	0.64789	0.31205999999999995
AB_one	ncbi367110	AB_one|NEUCR@367110@1|Q7S913|1	0.82199	0.9906999999999999
AB_one_2	ncbi81824	AB_one_2|MONBE@81824@1|A9UY30|1	0.9395199999999999	0.7356499999999999
AB_one_2	ncbi7227	AB_one_2|DROME@7227@1|Q9VXT2|1	0.9311499999999999	0.5563899999999999
```
And inside the `core_orthologs` folder:
```
(bionf) vinh@leonard:/share/project/vinh/test/fdog/fdogs/huangziyan$ ls -la AB_dir.combined/core_orthologs/
total 16
drwxr-xr-x 4 vinh users 4096 Mar 14 12:17 .
drwxr-xr-x 3 vinh users 4096 Mar 14 12:35 ..
drwxr-xr-x 4 vinh users 4096 Mar 14 12:22 AB_one
drwxr-xr-x 4 vinh users 4096 Mar 14 12:17 AB_one_2
(bionf) vinh@leonard:/share/project/vinh/test/fdog/fdogs/huangziyan$ ls -la AB_dir.combined/core_orthologs/AB_one_2/
total 24
drwxr-xr-x 4 vinh users 4096 Mar 14 12:17 .
drwxr-xr-x 4 vinh users 4096 Mar 14 12:17 ..
-rw-r--r-- 1 vinh users 2411 Mar 14 12:17 AB_one_2.aln
-rw-r--r-- 1 vinh users 1871 Mar 14 12:17 AB_one_2.fa
drwxr-xr-x 3 vinh users 4096 Mar 14 12:11 fas_dir
drwxr-xr-x 2 vinh users 4096 Mar 14 12:11 hmm_dir
```

**Reply by huangziyan11111** [reply]
Thank you for your reply. However, I have the same problem. The problem is caused by fas, because I found it's OK if I add this parameter "--fasoff" and run like you do.

**Reply by trvinh** [reply]
I cannot imagine that it has anything to do with FAS 🤔 I had only the problem with the "dots" in the input filenames.
Could you please try the v0.0.52 and let me know if it works?

**Reply by trvinh** [reply]
@huangziyan11111 can you please show me the sequence ID of your input files? Many thanks!

**Comment by trvinh** [comment]
Hi @huangziyan11111 
with the new version (v0.0.52) you can have "dots" in the input filenames. They will be automatically replaced by "_" during the processing.
Hope it works for you!
Best,
Vinh
